src/kernel.cr

```diff
@@ -217,23 +217,31 @@ end
 # Flag     | Applies to    | Meaning
 # ---------+---------------+--------------------------------------------
 # space    | bdiouxX       | Add a leading space character to
-#          | aAeEfgG       | non-negative numbers.
-#          | (numeric fmt) | For o, x, X, b, use
-#          |               | a minus sign with absolute value for
-#          |               | negative values.
+#          | aAeEfgG       | non-negative numbers. Has no effect if the plus
+#          | (numeric fmt) | flag is also given.
+# ---------+---------------+-----------------------------------------
+# #        | boxX          | Use an alternative format.
+#          | aAeEfgG       | For o, x, X, b, prefix any non-zero result with
+#          |               | 0o, 0x, 0X, or 0b respectively.
```

Comment:
Both C and Ruby omit the prefix for zero. I don't know why, but my guess is if the prefix is always needed then one can always write `0x%x` instead of `%#x`.